### PR TITLE
Issue #4 - sealed classes

### PR DESCRIPTION
**Pull request by bernerbrau**

### CODE REVIEW

**Comment by ntrrgc** [comment]
This is too much complex.

**Comment by ntrrgc** [comment]
It's hard to follow what has changed and what hasn't even using `git diff -w`.

**Comment by bernerbrau** [comment]
Yeah, I moved too much around. Let me revert and try again.

**Comment by ntrrgc** [comment]
My worry is that the patch to implement this feature is so big and far reaching that it becomes not worth it for maintainability sake.

**Comment by ntrrgc** [comment]
This was my attempt. It does not handle non-nested sealed class children, but it's much simpler. https://github.com/ntrrgc/ts-generator/commit/a12e9f2f1ece6bec97ac4c2ea4450003534e9557

**Comment by bernerbrau** [comment]
Right, agreed. In addition to your initial concerns, I realized that
typescript's type erasure makes it difficult at runtime to determine which
member of the union is being used; hence why I found it necessary to
include code generation that does runtime type validation. To do this, I
had to make the class visitor repeatable, and that was enough to confuse
git diff beyond hopefulness.

So it may simply be that sealed classes are too complicated to be worth
including. :(
